common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -142,6 +142,16 @@
      */
     private static final String PRIVATE_KEY_NOT_FOUND = "Not an instance of a PrivateKeyEntry";
 
+    /**
+     * Error message from underlying KeyStore that StrongBox HAL is unavailable.
+     */
+    public static final String STRONG_BOX_UNAVAILABLE_EXCEPTION = "StrongBoxUnavailableException";
+
+    /**
+     * Error message from underlying KeyStore that
+     */
+    public static final String FAILED_TO_GENERATE_ATTESTATION_CERTIFICATE_CHAIN = "Failed to generate attestation certificate chain";
```

Comment:
Is this documented somewhere?

Reply:
Sadly, [no ](https://developer.android.com/reference/android/security/keystore/KeyGenParameterSpec.Builder#setAttestationChallenge(byte[])) -- the documentation gives no indication that this can fail on N+ devices when used with RSA; however [this error](https://android.googlesource.com/platform/frameworks/base/+/ada0fc1/keystore/java/android/security/keystore/AndroidKeyStoreKeyPairGeneratorSpi.java#601) is cropping up on certain devices